libavfilter/dnn/dnn_backend_torch.cpp

```diff
@@ -0,0 +1,535 @@
+    request->output = NULL;
+    return request;
+}
+
+DNNModel *ff_dnn_load_model_th(const char *model_filename, DNNFunctionType func_type, const char *options, AVFilterContext *filter_ctx)
+{
+    DNNModel *model = NULL;
+    THModel *th_model = NULL;
+    THRequestItem *item;
+    THContext *ctx = &th_model->ctx;
+
+    model = (DNNModel *)av_mallocz(sizeof(DNNModel));
+    if (!model) {
+        return NULL;
+    }
+
+    th_model = (THModel *)av_mallocz(sizeof(THModel));
+    if (!th_model) {
+        av_freep(&model);
+        return NULL;
+    }
+
+    try {
+        th_model->jit_model = torch::jit::load(model_filename);
+    } catch (const c10::Error& e) {
+        goto fail;
+    }
+
+    th_model->ctx.c_class = &dnn_th_class;
+    model->options = options;
+    if (model->options) {
+        av_log((void *)ctx, AV_LOG_WARNING, "Current LibTorch backend do not support any options, \
+                                               all the options will be ignored\n");
+    }
+
+    th_model->request_queue = ff_safe_queue_create();
+    if (!th_model->request_queue) {
+        goto fail;
+    }
+
+    item = (THRequestItem *)av_mallocz(sizeof(THRequestItem));
+    if (!item) {
+        goto fail;
+    }
+    item->lltask = NULL;
+    item->infer_request = th_create_inference_request();
+    if (!item->infer_request) {
+        av_log(NULL, AV_LOG_ERROR, "Failed to allocate memory for TensorFlow inference request\n");
+        goto fail;
+    }
+    item->exec_module.start_inference = &th_start_inference;
+    item->exec_module.callback = &infer_completion_callback;
+    item->exec_module.args = item;
+
+    if (ff_safe_queue_push_back(th_model->request_queue, item) < 0) {
+        goto fail;
+    }
+
+    th_model->task_queue = ff_queue_create();
+    if (!th_model->task_queue) {
+        goto fail;
+    }
+
+    th_model->lltask_queue = ff_queue_create();
+    if (!th_model->lltask_queue) {
+        goto fail;
+    }
+
+    th_model->model = model;
+    model->model = th_model;
+    model->get_input = &get_input_th;
+    model->get_output = &get_output_th;
+    model->options = NULL;
+    model->filter_ctx = filter_ctx;
+    model->func_type = func_type;
+    return model;
+
+fail:
+    destroy_request_item(&item);
+    ff_queue_destroy(th_model->task_queue);
+    ff_queue_destroy(th_model->lltask_queue);
+    ff_safe_queue_destroy(th_model->request_queue);
+    av_freep(&th_model);
+    av_freep(&model);
+    av_freep(&item);
+    return NULL;
+}
+
+static void avframe_to_tensordata(AVFrame *frame, torch::Tensor *tensor)
+{
+    int height = frame->height;
+    int width = frame->width;
+    int size = height * width;
+    uint8_t **src = frame->data;
+    int stride = frame->linesize[0];
+    uint8_t *src_line;
+    float *dst = NULL;
+    dst = (float*)malloc(size * 3 * sizeof(float));
+    if (!dst) {
+        delete(tensor);
+        tensor = NULL;
+        return;
+    }
+    //convert between packed and planner RGB24
```

Comment:
seems we have simliar function in swscale, gbr24ptopacked24. 
could we use swscale do this?

Reply:
Will add code to support function like gbr24ptopacked24

Reply:
> seems we have simliar function in swscale, gbr24ptopacked24. could we use swscale do this?

Will add code to support function like gbr24ptopacked24